packages/xstate-react/src/fsm.ts

```diff
@@ -10,6 +10,17 @@ export function useMachine<TC, TE extends EventObject = AnyEventObject>(
   StateMachine.Service<TC, TE>['send'],
   StateMachine.Service<TC, TE>
 ] {
+  if (process.env.NODE_ENV !== 'production') {
+    const [initialMachine] = useState(stateMachine);
```

Comment:
Hmm, can we do this without using `useState`? Perhaps `useEffect`

Reply:
What problem do you see with useState here? Its either this or useConstant but as this is dev path i’ve chosen state as the simpler one. We need some “memory cell” to keep the initial value, so either state or ref.